draft_1_proposal_instructions.md

```diff
@@ -0,0 +1,144 @@
+In the future, the contents of the "SNS lifecycle" section in ./README.md would
+be replaced with the content in this file (minus this preface, ofc).
+
+_This section assumes that you have successfully completed the instructions in
+the [Bootstrapping a testing environment via Docker] section above._
+
+This section explains how to use the aforementioned testing environment to
+perform SNS decentralization of your own dapp running in the local `dfx` server
+test environment.
+
+The commands mentioned here are to be run from within the Docker container.
+
+'run_basic_scenario.sh` also performs these steps. Therefore, reading that
+script may help you understand the steps described in this section. Similarly,
+reading the scripts mentioned here will help you understand the procedure in
+greater detail.
+
+0. Restore files to a clean state like so:
+
+   ```bash
+   ./cleanup.sh
+   ```
+
+1. Deploy your dapp onto the local `dfx` server the way you normally would. You
+   can find your dapp repo under the path `/dapp` in the Docker container. Note
+   that the `dfx` server is using port 8080.
+
+   Alternatively, you can deploy the [example dapp] like so:
+
+   ```bash
+   ./deploy_test_canister.sh
+   ```
+
+   [example dapp]: https://github.com/dfinity/sns-testing#test-canister
+
+2. Craft your own SNS configuration file. We recommend that you use
+   example_sns_init.yaml as a guide.
+
+   **TODO**: Write a script that does this for the example dapp. The
+     main things it needs to do are:
+
+   1. Replace aaaaa-aa with whatever `dfx identity get-principal`
+      prints.
+
+   2. Replace the dapp canister principal ID with whatever
+      deploy_test_canister.sh reports.
+
+   3. I don't think there's anything else, but I could be wrong...
+
+3. Give control of your dapp canisters to NNS:
+
+   ```bash
+   ./let_nns_control_dapp.sh
+   ```
+
+4. Create an NNS proposal to create an SNS that will control your dapp
+   canisters:
+
+   ```bash
+   ./propose_sns.sh
+   ````
+
+   The proposal will pass right away, because it is made by a neuron that has an
+   overwhelming amount of voting power.
+
+   After a few minutes, you should see a new SNS instance in the [Launchpad]
+   section of the NNS Web app.
+
+   [Launchpad]: http://qsgjb-riaaa-aaaaa-aaaga-cai.localhost:8080/launchpad
+
+5. Optional: Upgrade your dapp via SNS proposal.
+
+   If you are going through these instructions using the example dapp rather
+   than your own dapp, this step can be done as follows:
+
+   ```bash
+   ./upgrade_test_canister.sh "Swap is taking place."
+   ```
+
+6. **TODO**: There is a 24-48 hour delay between proposal execution and the
```

Comment:
We're already pulling sns_governance_test so we could similarly pull sns_swap_test if you create such a version of the sns_swap canister with some checks disabled by a compile-time flag.